skimage/measure/fit.py

```diff
@@ -735,8 +736,8 @@ def ransac(data, model_class, min_samples, residual_threshold,
     >>> x = xc + a * np.cos(t)
     >>> y = yc + b * np.sin(t)
     >>> data = np.column_stack([x, y])
-    >>> np.random.seed(seed=1234)
-    >>> data += np.random.normal(size=data.shape)
+    >>> rng = np.random.default_rng()
+    >>> data += rng.normal(size=data.shape)
```

Comment:
Can we perhaps continue setting the seed here (and below) so that the doctest can still be run? In some cases it's unclear whether things are suitably tested without the doctest...!

Reply:
Do you have a CI job for the doctests?

We had some discussions for SciPy (and still are having some on Astropy) and we came up with a monkeypatching of the line `>>> rng = np.random.default_rng()` to include a seed. To me this is no practical but it works. Here it does not seems that you have much examples which would really need a seed. Hence I would advice not to use any. And if there are things that are tested in the examples which are not in the tests, then we should simply move these to tests. In the end it's double the work for you maintainers to maintain 2 test suites otherwise.

Reply:
Yes, the doctests are configured run by default. imho the double effort is in the other direction — if we add tests in addition to doctests (which no longer become doctests), then we have to maintain them in two places.

Reply:
I was more thinking about moving than duplicating. IMO doctests should only serve to check that examples are still aligned with the implementation.

Anyway, what I can propose is to either put a seed when really necessary and comment with a `DO NOT COPY`. So that you could still use them for testing. For the seed itself, I would not use the same for all examples and I would get it using `np.random.SeedSequence`. Would that be ok?

Reply:
Yup, that all sounds fine! Sometimes an example usage is close enough to what a test would be that you can kill two birds with one stone — and we do this liberally at skimage. :blush: :grimacing:

Reply:
@hmaarrfk we had this discussion. So in the example you noted, I missed to do this. Let me know what you choose so I modify it accordingly if needed.